modules/javafx.controls/src/test/java/test/javafx/scene/control/TreeCellTest.java

```diff
@@ -888,6 +889,85 @@ public void testCommitEditMustNotFireCancel() {
         assertEquals("commit must not have fired editCancel", 0, events.size());
     }
 
+//------------- JDK-8187309: fix editing mechanics to comply to specification
+
+    @Test
+    public void testTreeHasDefaultCommitHandler() {
+        assertNotNull("treeView must have default commit handler", tree.getOnEditCommit());
+    }
+
+    @Test
+    public void testDefaultCommitUpdatesData() {
+        TreeItem<String> editingItem = setupForEditing(cell);
+        tree.edit(editingItem);
+        String value = "edited";
+        cell.commitEdit(value);
+        assertEquals("value committed", value, editingItem.getValue());
+    }
+
+    @Test
+    public void testDefaultCommitUpdatesCell() {
+        TreeCell<String> cell = TextFieldTreeCell.forTreeView().call(tree);
+        TreeItem<String> editingItem = setupForEditing(cell);
+        tree.edit(editingItem);
+        String value = "edited";
+        cell.commitEdit(value);
+        assertEquals("cell text updated to committed value", value, cell.getText());
+    }
+
+    @Test
+    public void testDoNothingCommitHandlerDoesNotUpdateData() {
+        TreeItem<String> editingItem = setupForEditing(cell);
+        String oldValue = editingItem.getValue();
+        // do nothing handler
+        tree.setOnEditCommit(e -> {});
+        tree.edit(editingItem);
+        String value = "edited";
+        cell.commitEdit(value);
+        assertEquals("edited value must not be committed", oldValue, editingItem.getValue());
+    }
+
+    @Ignore("JDK-8187314")
+    @Test
+    public void testDoNothingCommitHandlerDoesNotUpdateCell() {
+        TreeCell<String> cell = TextFieldTreeCell.forTreeView().call(tree);
+        TreeItem<String> editingItem = setupForEditing(cell);
+        String oldValue = editingItem.getValue();
+        // do nothing handler
+        tree.setOnEditCommit(e -> {});
+        tree.edit(editingItem);
+        String value = "edited";
+        cell.commitEdit(value);
+        assertEquals("cell text must not have changed", oldValue, cell.getText());
+    }
+
+
+    /**
+     * Sets tree editable, configures the given cell for editing in tree at index 1
+     * and returns the treeItem at that index.
+     */
+    private TreeItem<String> setupForEditing(TreeCell<String> editingCell) {
+        tree.setEditable(true);
+        editingCell.updateTreeView(tree);
+        editingCell.updateIndex(1);
+        return editingCell.getTreeItem();
+    }
+
+    /**
+     * Test test setup.
```

Comment:
Minor: I would rephrase that a bit. Something like:
`Tests the cell editing setup`.

Reply:
oops .. forgot to change this (and the other comment) before integration - sry